### PR TITLE
Test status page fixes

### PR DESCRIPTION
**Pull request by islemaster**
Some little across-the-board improvements to the UI test status page.
- Inject the test server origin into the generated test status page, so when asking for updates it can use full URLs instead of origin-relative URLs.  This puts us a step closer to serving the test status page from anywhere (like S3) though we're not quite there yet - we haven't enabled cross-origin requests, and it still needs to access its JavaScript somehow.
- Using the test dashboard on localhost depends on a `dashboard/public/ui_test` symlink being set up.  This now happens on `rake build:dashboard` in all environments except production (previously it was just part of our UI test setup in ci_build).
- Being more ES6-y in general (the status page assumes you're running a recent version of Chrome).
- Other changes inline.

### CODE REVIEW

File: dashboard/test/ui/test_status.js [hunk 1]
```diff
@@ -52,8 +53,8 @@ function Test(fromRow) {
   /** @type {number} in seconds */
   this.duration = 0;
 
-  /** @private {string} */
-  this.status_ = STATUS_PENDING;
+  /** @type {string} */
+  this.status = STATUS_PENDING;
```
Comment (islemaster) [inline dashboard/test/ui/test_status.js:58]: We were using this outside the class; renamed and made non-private.

File: dashboard/test/ui/test_status.js [hunk 2]
```diff
@@ -154,8 +155,8 @@ rows.forEach(row => {
 function testFromS3Key(key) {
   var re = /[^/]+\/([^_]+)_(.*)_output\.html/i;
   var result = re.exec(key);
-  var browser = result[1]; // TODO: Handle slashes
```
Comment (islemaster) [inline dashboard/test/ui/test_status.js:157]: TODO was addressed in a previous PR, just didn't get removed.

File: dashboard/test/ui/test_status.js [hunk 3]
```diff
@@ -171,7 +172,7 @@ function testFromS3Key(key) {
   // unlikely.
   var test = tests[browser][feature];
   while (!test && feature.indexOf('_') >= 0) {
-    feature = feature.replace('_', '/'); // Replaces first underscore only.
+    feature = feature.replace(/_/, '/'); // Replaces first underscore only.
```
Comment (islemaster) [inline dashboard/test/ui/test_status.js:176]: This comment was a lie (and so is the documentation on MDN apparently) - when using a `string` needle the replacement was global even though I intended it to be only the first occurrence.  Using a `RegExp` needle it should be just the first one, since I'd have to pass the `/g` global flag to replace all.

File: dashboard/test/ui/test_status.js [hunk 4]
```diff
@@ -306,4 +307,5 @@ function toggleAutoRefresh() {
 refreshButton.onclick = refresh;
 autoRefreshButton.onclick = toggleAutoRefresh;
 enableAutoRefresh();
+updateProgress();
```
Comment (islemaster) [inline dashboard/test/ui/test_status.js]: Calling this on page load makes the progress bars show up immediately, where before they would be invisible until there was progress.

File: dashboard/test/ui/test_status.js (hunk 3 above)
Comment (Hamms) [inline dashboard/test/ui/test_status.js:176]: Huh. I could have sworn string needles only ever replaced the first instance; could this be an ES6 thing? Perhaps babel transpiling something it shouldn't?

File: dashboard/test/ui/test_status.js [hunk 5]
```diff
@@ -210,14 +211,26 @@ function renderBrowserProgress(browser, progress) {
...
-  let pendingPercent = Math.max(Math.floor(pendingCount * 100 / totalCount), pendingCount > 0 ? 1 : 0);
+  let successPercent = Math.max(Math.floor(successCount * 1000 / totalCount)/10, successCount > 0 ? 0.1 : 0);
+  let failurePercent = Math.max(Math.floor(failureCount * 1000 / totalCount)/10, failureCount > 0 ? 0.1 : 0);
+  let pendingPercent = Math.max(Math.floor(pendingCount * 1000 / totalCount)/10, pendingCount > 0 ? 0.1 : 0);
   let leftover = 100 - (successPercent + failurePercent + pendingPercent);
-  if (pendingCount > 0) {
-    pendingPercent += leftover;
-  } else {
-    successCount += leftover;
+  if (leftover > 0) {
```
Comment (Hamms) [inline dashboard/test/ui/test_status.js:219]: Could you explain the motivation for this change?

File: dashboard/test/ui/test_status.js [hunk 6]
```diff
@@ -267,7 +281,7 @@ function refresh() {
   };
   let lastRefreshEpochSeconds = Math.floor(lastRefreshTime.getTime()/1000);
   let newTime = new Date();
-  fetch(`/api/v1/test_logs/${gitBranch}/since/${lastRefreshEpochSeconds}`, { mode: 'no-cors' })
+  fetch(`${API_ORIGIN}/api/v1/test_logs/${GIT_BRANCH}/since/${lastRefreshEpochSeconds}`, { mode: 'no-cors' })
```
Comment (Hamms) [inline dashboard/test/ui/test_status.js]: Should `"api/v1/test_logs"` be pulled out into a constant?

**Comment by Hamms** [comment]
This LGTM, although with the caveat that I'm entirely too unfamiliar with this element to notice if there was something of concern

File: dashboard/test/ui/test_status.js (hunk 3 above)
Comment (islemaster) [inline dashboard/test/ui/test_status.js:176]: This file doesn't get preprocessed in any way, so it shouldn't be an ES6 or transpiling thing.  I just noticed while doing some debugging.

I like using regex everywhere I can anyway :trollface:

File: dashboard/test/ui/test_status.js (hunk 5 above)
Comment (islemaster) [inline dashboard/test/ui/test_status.js:219]: There was a bug before where rounding could add up to a percent greater than 100%, which would result in a fairly arbitrary reduction from the pending percentage, which could display 100% complete before we're actually done.  I've added a single place of precision to the progress bars, and updated the rounding correction to err on the side of incompleteness.

Example bad case before:

```
// Actual counts
successCount = 199
failureCount = 1
pendingCount = 1
// Calculated percentages before accounting for leftover
successPercent = 99
failurePercent = 1
pendingPercent = 1
leftover = -1
```

 By the old logic we would have applied `pendingPercent += leftover` so pendingPercent drops to 0 even though it's nonzero.  The final success/failure/pending result was 99 / 1 / 0. This gave a misleading indication of test completeness:
screenshot from 2016-06-23 11-25-46

Here's how the new logic handles this edge case (modified to show edge case in increased granularity):

```
// Actual counts
successCount = 1999
failureCount = 1
pendingCount = 1
// Calculated percentages before accounting for leftover
successPercent = 99.9
failurePercent = 0.1
pendingPercent = 0.1
leftover = -0.1
```

The new logic notes that the leftover is negative and the successes are nonzero, so it deducts from the success percentage first, giving a result of 99.8 / 0.1 / 0.1, correctly indicating that the tests are not done running.

File: dashboard/test/ui/test_status.js (hunk 6 above)
Comment (islemaster) [inline dashboard/test/ui/test_status.js]: Sure, why not.
